sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -499,27 +502,70 @@ class kernel_bundle_impl {
     if (MLanguage == syclex::source_language::sycl_jit) {
       // Build device images via the program manager.
       const std::string &SourceStr = std::get<std::string>(MSource);
+      std::ostringstream SourceExt;
```

Comment:
Why is the streaming of the kernel names onto the source string done here in kernel_bundle_impl.hpp?  To me it seems more like an implementation detail that should occur in the `SYCL_JIT_to_SPIRV` function, no?

On the other hand, I realize this `kernel_bundle_impl` constructor is only for the SYCL_JIT kernel bundles, and streaming like this could be fairly characterized as "marshalling".  

But, still, it seems a little "under the covers".